src/reducers/updateThoughts.js

```diff
@@ -11,22 +10,24 @@ import {
 
 /**
  * Updates thoughtIndex and contextIndex with any number of thoughts
- * SIDE EFFECT: Sync thoughts here before thoughtIndexUpdates and contextIndexUpdates get merged. By the time there is a new state it is too late, and I want to avoid moving everything into impure action-creators.
  */
 export default (state, { thoughtIndexUpdates, contextIndexUpdates, recentlyEdited, contextChain }) => {
 
   const thoughtIndex = mergeUpdates(state.thoughtIndex, thoughtIndexUpdates)
   const contextIndex = mergeUpdates(state.contextIndex, contextIndexUpdates)
   const recentlyEditedNew = recentlyEdited || state.recentlyEdited
 
-  setTimeout(() => {
-    sync(thoughtIndexUpdates, contextIndexUpdates, { recentlyEdited: recentlyEditedNew })
-  })
+  // updates are queued, detected by the syncQueue middleware, and sync'd with the local and remote stores
+  const syncQueueNew = {
+    thoughtIndexUpdates: { ...syncQueueOld.thoughtIndexUpdates, ...thoughtIndexUpdates },
+    contextIndexUpdates: { ...syncQueueOld.contextIndexUpdates, ...contextIndexUpdates },
+  }
```

Comment:
Here `syncQueueOld` is not defined. Maybe you forgot to get old queue from the state.

Reply:
Thanks, you're right